### PR TITLE
adds finally logs in pipelinerun

### LINKED ISSUE #1054: PipelineRun logs do not include finally task logs
# Versions and Operating System

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-13T11:52:32Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
  Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.11+IKS", GitCommit:"c33b093e937d1f9eee340f9bddfa14e1e54ad1a4", GitTreeState:"clean", BuildDate:"2020-06-17T21:37:54Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
  ```

- Tekton Pipeline version:

  **Output of `tkn version`:**

  ```
  Client version: 0.10.0
  Pipeline version: devel
  Triggers version: unknown
  ```

  **Output of `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

  ```v0.14.0```

- Operating System:

  ```
  Mac OS 10.15.5
  ```

# Expected Behavior

`tkn pipelinerun logs ...` should include the logs from all tasks including the [`finally` tasks](https://github.com/tektoncd/pipeline/blob/v0.14.0/docs/pipelines.md#adding-finally-to-the-pipeline).

# Actual Behavior

`tkn pipelinerun logs ...` does not include the logs of the [`finally` tasks](https://github.com/tektoncd/pipeline/blob/v0.14.0/docs/pipelines.md#adding-finally-to-the-pipeline).

# Steps to Reproduce the Problem

## 1. Run pipeline `exit_handler.yaml` pipeline:

```Bash
cat <<EOF > exit_handler.yaml

apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  name: exit-handler
spec:
  pipelineSpec:
    finally:
    - name: echo-on-exit
      taskSpec:
        steps:
        - args:
          - echo "finally"
          command:
          - sh
          - -c
          image: library/bash:4.4.23
          name: main
    tasks:
    - name: print-msg
      taskSpec:
        steps:
        - args:
          - echo "printing a message"
          command:
          - sh
          - -c
          image: library/bash:4.4.23
          name: main
EOF

kubectl apply -f exit_handler.yaml

# pipelinerun.tekton.dev/exit-handler created

tkn taskrun list

# NAME                                               STARTED        DURATION     STATUS
# exit-handler-echo-on-exit-5n9dk                    12 hours ago   4 seconds    Succeeded
# exit-handler-print-msg-xcl6p                       12 hours ago   4 seconds    Succeeded


```

## 2. Get the pipelinerun logs with `tkn`:

```Bash
tkn pipelinerun logs exit-handler 

# [print-msg : main] hi there
```

**Expected:**

```Bash
tkn pipelinerun logs exit-handler 

# [print-msg : main] hi there
# [echo-on-exit : main] finally
```

## 3. Get the pipelinerun logs with `kubectl`:

```Bash
for POD_NAME in $(kubectl get pipelineruns exit-handler  -o jsonpath='{.status.taskRuns.*.status.podName}'); do \
    kubectl logs $POD_NAME --all-containers ; \
done;

# finally
# hi there
```

### PR DESCRIPTION
**Pull request by pradeepitm12**
closes #1054

this adds finally logs in pipelinerun logs command in follow and
non follow mode

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/pipelinerun/logs_test.go [hunk 1]
```diff
@@ -2450,6 +2518,627 @@ func TestLog_pipelinerun_only_one_v1beta1(t *testing.T) {
...
+
+	expectedLogs := []string{
+		"[output-task : writefile-step] wrote a file1\n",
+		"[finally-task : finally-step] Finally\n",
+	}
+	expected := strings.Join(expectedLogs, "\n") + "\n"
+	test.AssertOutput(t, expected, output)
+}
+
+/*func TestPipelinerunLog_with_follow_mode_finally_task_v1beta1(t *testing.T) {
```
Comment (vdemeester) [inline pkg/cmd/pipelinerun/logs_test.go:2746]: We should removed this code if commented :wink:

Comment (vdemeester) [reply pkg/cmd/pipelinerun/logs_test.go:2746]: And write an e2e tests instead for it :stuck_out_tongue_closed_eyes:

Comment (pradeepitm12) [reply pkg/cmd/pipelinerun/logs_test.go]: Yeh :stuck_out_tongue_closed_eyes:
Changed this to e2e

**Comment by pradeepitm12** [comment]
need to ignore finally related test for v0.10.0

**Comment by danielhelfand** [comment]
> need to ignore finally related test for v0.10.0

This is what I did in #1066: https://github.com/tektoncd/cli/blob/209581e52521aa36449e97837741fc256e355c5f/test/e2e/cli.go#L218

And used in test:
https://github.com/tektoncd/cli/blob/209581e52521aa36449e97837741fc256e355c5f/test/e2e/task/start_test.go#L130

**Comment by pradeepitm12** [comment]
> > need to ignore finally related test for v0.10.0
> 
> This is what I did in #1066:
> 
> https://github.com/tektoncd/cli/blob/209581e52521aa36449e97837741fc256e355c5f/test/e2e/cli.go#L218
> 
> And used in test:
> https://github.com/tektoncd/cli/blob/209581e52521aa36449e97837741fc256e355c5f/test/e2e/task/start_test.go#L130

Thank you:smiley:  this is helpful

File: test/e2e/pipelinerun/pipelinerun_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	if tkn.CheckVersion("Pipeline", "v0.10.2") {
+		t.Skip("Skip test as pipeline v0.10.2 doesn't support finally")
+	}
+
+	t.Logf("Creating pipelinerun in namespace: %s", namespace)
+	e2e.Assert(t, kubectl.Create(e2e.ResourcePath("pipelinerun-with-finally.yaml")), icmd.Success)
+
+	t.Run("Pipelinerun logs with finally  "+namespace, func(t *testing.T) {
```
Comment (danielhelfand) [inline test/e2e/pipelinerun/pipelinerun_test.go:45]: Remove nested `t.Run`:

```go
t.Run("Pipelinerun logs with finally  "+namespace, func(t *testing.T) {
	res := tkn.Run("pipelinerun", "logs", "exit-handler", "-f")
	log := res.Stdout()
	s := []string{
		"[print-msg : main] printing a message\n",
		"[echo-on-exit : main] finally\n",
	}
	expected := strings.Join(s, "\n") + "\n"
	e2e.AssertOutput(t, expected, log)
})
```

**Pull request merged**